plugin/path_impersonated_account_test.go

```diff
@@ -53,6 +53,49 @@ func TestPathImpersonate_Basic(t *testing.T) {
 	testImpersonateDelete(t, td, impersonateName)
 }
 
+func TestPathImpersonate_TTL(t *testing.T) {
+	impersonateName := "test-impersonated-basic"
+
+	td := setupTest(t, "0s", "2h")
+	defer cleanupImpersonate(t, td, impersonateName, util.StringSet{})
+
+	sa := createServiceAccount(t, td, impersonateName)
+	defer deleteServiceAccount(t, td, sa)
+
+	// 1. Read should return nothing
+	respData := testImpersonateRead(t, td, impersonateName)
+	if respData != nil {
+		t.Fatalf("expected impersonate account to not exist initially")
+	}
+
+	// 2. Create new impersonated account with a TTL of 4 hours
+	// (longer than MaxTTL of 2 horus)
+	const ttl = 60 * 60 * 4
+	testImpersonateCreate(t, td, impersonateName,
+		map[string]interface{}{
+			"service_account_email": sa.Email,
+			"token_scopes":          []string{iam.CloudPlatformScope},
+			"ttl":                   ttl,
+		})
+
+	// 3. Read impersonated account
+	respData = testImpersonateRead(t, td, impersonateName)
+	if respData == nil {
+		t.Fatalf("expected impersonate account to have been created")
+	}
+
+	verifyReadData(t, respData, map[string]interface{}{
+		"service_account_email":   sa.Email,
+		"service_account_project": sa.ProjectId,
+		"ttl":                     ttl,
```

Comment:
should the returned TTL here be 2h? It looks like the expected `ttl` is still 4h but the comment on line 71 indicates that MaxTTL should be 2h.

Reply:
So this is just validating the CRUD for the Impersonated Account Config.  Based on the earlier conversation, we decided to clamp the Token Lifetime but not the CRUD data for Impersonated Account based on the config of the Backend.

- https://github.com/DataDog/vault-plugin-secrets-gcp/pull/2#discussion_r908830866

The reason being the lifecycles of the backend config and impersonated account config can be different, and clamping the impersonated account config on write or read has other implications for various workflows.